File: article/src/main/java/com/ruzhan/awaker/article/db/converter/RoomDataConverter.java
package com.ruzhan.awaker.article.db.converter;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.ruzhan.awaker.article.model.Comment;
import com.ruzhan.awaker.article.model.News;
import com.ruzhan.awaker.article.model.Special;

import java.util.Date;
import java.util.List;


public class RoomDataConverter {
    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

    @TypeConverter
    public static List<Special> toSpecialList(String json) {
        return new Gson().fromJson(json, new TypeToken<List<Special>>() {
        }.getType());
    }

    @TypeConverter
    public static String toSpecialListJson(List<Special> specialList) {
        return new Gson().toJson(specialList);
    }

    @TypeConverter
    public static List<Comment> toCommentList(String json) {
        return new Gson().fromJson(json, new TypeToken<List<Comment>>() {
        }.getType());
    }

    @TypeConverter
    public static String toCommentListJson(List<Comment> commentList) {
        return new Gson().toJson(commentList);
    }

    @TypeConverter
    public static List<News> toNewsList(String json) {
        return new Gson().fromJson(json, new TypeToken<List<News>>() {
        }.getType());
    }

    @TypeConverter
    public static String toNewsListJson(List<News> newsList) {
        return new Gson().toJson(newsList);
    }
}

File: article/src/main/java/com/ruzhan/awaker/article/db/converter/RoomDataConverter.kt
package com.ruzhan.awaker.article.db.converter

import android.arch.persistence.room.TypeConverter

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.ruzhan.awaker.article.model.Comment
import com.ruzhan.awaker.article.model.News
import com.ruzhan.awaker.article.model.Special

import java.util.Date


object RoomDataConverter {
    @TypeConverter
    @JvmStatic
    fun toDate(timestamp: Long?): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    @TypeConverter
    @JvmStatic
    fun toTimestamp(date: Date?): Long? {
        return date?.time
    }

    @TypeConverter
    @JvmStatic
    fun toSpecialList(json: String): List<Special> {
        return Gson().fromJson(json, object : TypeToken<List<Special>>() {

        }.type)
    }

    @TypeConverter
    @JvmStatic
    fun toSpecialListJson(specialList: List<Special>): String {
        return Gson().toJson(specialList)
    }

    @TypeConverter
    @JvmStatic
    fun toCommentList(json: String): List<Comment> {
        return Gson().fromJson(json, object : TypeToken<List<Comment>>() {

        }.type)
    }

    @TypeConverter
    @JvmStatic
    fun toCommentListJson(commentList: List<Comment>): String {
        return Gson().toJson(commentList)
    }

    @TypeConverter
    @JvmStatic
    fun toNewsList(json: String): List<News> {
        return Gson().fromJson(json, object : TypeToken<List<News>>() {

        }.type)
    }

    @TypeConverter
    @JvmStatic
    fun toNewsListJson(newsList: List<News>): String {
        return Gson().toJson(newsList)
    }
}
